polus-feature-extraction-plugin/src/utility.py

```diff
@@ -0,0 +1,1035 @@
+                'equivalent_diameter': equivalent_diameter,
+                'euler_number': euler_number,
+                'major_axis_length': major_axis_length,
+                'minor_axis_length': minor_axis_length,
+                'solidity': solidity,
+                'mean_intensity': mean_intensity,
+                'max_intensity': max_intensity,
+                'min_intensity': min_intensity,
+                'median': median,
+                'mode': mode,
+                'standard_deviation': standard_deviation,
+                'skewness': skewness,
+                'entropy': entropy,
+                'kurtosis': kurtosis,
+                'neighbors': neighbors,
+                'maxferet': maxferet,
+                'minferet': minferet,
+                'polygonality_score': polygonality_score,
+                'hexagonality_score': hexagonality_score,
+                'hexagonality_sd': hexagonality_sd,
+                'all': all}
+      
+        #Calculate features given as input for all images
+        regions = measure.regionprops(self.label_image,self.intensity_image)#region properties
+        title = self.filenames#to pass the filename in csv
+        
+        for each_feature in self.feature:
+            label = [r.label for r in regions]#get labels list for all regions
+            feature_value = FEAT[each_feature](self.label_image,self.intensity_image)#dynamically call the function based on the features required
+            
+            #get all features
+            if each_feature  == 'all':
+                df=pd.DataFrame(feature_value)#create dataframe for all features
+                df = df.T#transpose
+                df.columns =['Area','Centroid row','Centroid column','Convex area','Eccentricity','Equivalent diameter','Euler number','Hexagonality score','Hexagonality sd','Kurtosis','Major axis length','Maxferet','Maximum intensity','Mean intensity','Median','Minimum intensity','Minferet','Minor axis length','Mode','Neighbors','Orientation','Perimeter','Polygonality score','Standard deviation','Skewness','Solidity']
+            else:    
+                df = pd.DataFrame({each_feature: feature_value})#create dataframe for features selected
+            self.df_insert = pd.concat([self.df_insert, df], axis=1)    
+        self.df_insert.insert(0, 'Image', title)#Insert filename as 1st column 
+        self.df_insert.insert(1, 'Label', label)#Insert label as 2nd column
+        self.df_insert.columns = map (lambda x: x.capitalize(),self.df_insert.columns)#Capitalize the first letter of header
+        
+        #save each cav file separately
+        if self.csv_file == 'csvmany':
+            csv_file= Df_Csv_multiple(self.df_insert, self.output_dir,title)
+            csv_final = csv_file.csvfilesave()
+        else:
+            self.df_csv = self.df_csv.append(self.df_insert)
+        return self.df_csv
+
+#class to save features for all images in same file    
+class Df_Csv_single(Analysis):
```

Comment:
Add class and method commenting according to the Google Python commenting specification.